Tests/Tactics/CSE.lean

```diff
@@ -15,38 +15,147 @@ We want to check that it correctly ees that there are:
 
 -/
 
-set_option trace.Tactic.cse.generalize true in
 set_option trace.Tactic.cse.summary true in
 /--
 warning: declaration uses 'sorry'
 ---
-info: [Tactic.cse.summary] CSE collected expressions: ⏎
-    y := { occs := 16, size := 1 }
-    y + y + (y + y) + (y + y + (y + y)) + (y + y + (y + y)) := { occs := 1, size := 23 }
-    y + y + (y + y) + (y + y + (y + y)) := { occs := 1, size := 15 }
-    y + y + (y + y) := { occs := 4, size := 7 }
-    x + x := { occs := 1, size := 3 }
-    x := { occs := 2, size := 1 }
-    x + x + (y + y + (y + y)) := { occs := 1, size := 11 }
-    y + y := { occs := 8, size := 3 }
-[Tactic.cse.generalize] ⏭️ Skipping y + y + (y + y) + (y + y + (y + y)) +
-      (y + y + (y + y)): Unprofitable { occs := 1, size := 23 } .
```

Comment:
I'm not sure how valuable having tests with tracing enabled is: I'm certainly not confident that I can review this diff and say for sure that the new trace does what it's supposed to. I imagine we might want to tweak the tracing pretty often, and if we just blindly accept the new trace as the expected output every time, the test won't catch any "bugs" in the new trace anyway

Reply:
@alexkeizer I agree, I see the current situation as a 'stop-gap', until we have the ability to look for patterns in guard messages. At least for me, it's easy to skim and see that the CSE is 'doing the right thing as per vibes', so I'd opt to keep it in.